utils/hoursUtils.js

```diff
@@ -0,0 +1,131 @@
+const airtableClient = require('../airtableClient')
+
+
+module.exports = {
+
+    getHoursInfo: function(periodsArray, hoursList) {
+        const hoursRecords = periodsArray.map( id => hoursList.find(x => x.id == id))
+        const hoursFields = hoursRecords.map(x => x.fields)
+        const hoursSummary = getHoursSummary(hoursFields)
+    
+        const datetimeNow = getDatetimeNow()
+        const todayDigit = datetimeNow.getDay()
+        const todayHours = hoursFields.find(period => period.open_weekday_digit == todayDigit)
+        const hoursWindow = todayHours ? parseTodayHours(todayHours, datetimeNow) : undefined
+    
+        if(hoursWindow) {
+            return {
+                ...hoursWindow,
+                hours: hoursSummary
+            }
+        } else {
+            return {
+                isOpenNow: false,
+                openingSoon: undefined,
+                closingSoon: undefined,
+                hours: hoursSummary
+            }
+        }
+    },
+    
+    transformHours: function(time) {
+        if(!isNaN(time)) {
+            const minutes = time[2] + time[3]
+            var hours = time[0] + time[1]
+            const ampm = hours >= 12 ? "pm" : "am"
+            hours = (hours % 12) || 12
+            return `${hours}:${minutes} ${ampm}`
+        } else {
+            return time
+        }
+    }
+    
+}
+
+getHoursSummary = function(hoursFields) {
+    const formattedHours = hoursFields.map(field => 
+        {
+            const closeTimeDigits = field.close_time_digits
+            const hoursObject = {
+                day: field.open_weekday,
+                day_digit: field.open_weekday_digit,
+                openTime: field.open_time,
+                open_time_digits: field.open_time_digits,
+                closeTime: field.close_time,
+                close_time_digits: closeTimeDigits ? closeTimeDigits : undefined,
+                hoursSummary: closeTimeDigits ? `${field.open_time} - ${field.close_time}` : "Open 24 hours"
+            }
+            return hoursObject
+        }
+    )
+
+    const weeklySchedule = [
```

Comment:
Currently only handles single-day, single period schedules. Will build in overnight and split shifts next.